src/api/procedures/investInSto.ts

```diff
@@ -110,25 +118,21 @@ export async function prepareInvestInSto(
 
   const portfolio = portfolioIdToPortfolio(fundingPortfolioId, context);
 
-  const { status, end, minInvestment, tiers, raisingCurrency } = await sto.details();
+  const {
+    status: { sale, timing },
+    minInvestment,
+    tiers,
+    raisingCurrency,
+  } = await sto.details();
 
   const [{ total: totalTokenBalance }] = await portfolio.getTokenBalances({
     tokens: [raisingCurrency],
   });
 
-  if (status !== StoStatus.Live) {
-    throw new PolymeshError({
-      code: ErrorCode.ValidationError,
-      message: 'The STO is not live',
-    });
-  }
-
-  const now = new Date();
-
-  if (end && now > end) {
+  if (sale !== StoSaleStatus.Live || timing !== StoTimingStatus.Started) {
```

Comment:
Should `balanceStatus` be `Available` too?

Reply:
Agreed

Reply:
Wait, that is already considered further down. The reason it's calculated apart and not derived from the status is the `maxPrice` parameter